src/jquery.uls.lcd.js

```diff
@@ -363,22 +362,20 @@
 		},
 
 		noResults: function () {
-			var $suggestions;
-			this.$noResults.removeClass( 'hide' );
-			this.$noResults.siblings( '.uls-lcd-region-section' ).addClass( 'hide' );
+			var $suggestions, $noResults;
 
-			// Only build the data once
-			if ( this.$noResults.find( '.uls-lcd-region-title' ).length ) {
-				return;
-			}
+			this.$element.addClass( 'uls-no-results' );
+			$noResults = this.$element.children( '.uls-no-results-view' );
 
 			$suggestions = this.buildQuicklist().clone();
-			$suggestions.removeClass( 'hide' );
-			$suggestions.find( 'h3' )
+			$suggestions
+				.removeClass( 'hide' )
+				.find( 'h3' )
 				.data( 'i18n', 'uls-no-results-suggestion-title' )
 				.text( 'You may be interested in:' )
 				.i18n();
-			this.$noResults.find( 'h2' ).after( $suggestions );
+
+			$noResults.find( 'h2' ).after( $suggestions );
```

Comment:
This looks very unreliable now that $noResults is completely out of our control.

Reply:
I moved this to the default handler for no-results